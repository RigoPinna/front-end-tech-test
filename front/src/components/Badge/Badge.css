.__badge {
    align-items: center;
    background:#E1422D;
    border-radius:100%;
    color: #FFFFFF;
    display: flex;
    font-size:12px;
    justify-content: center;
    height: 18px;
    min-width: 18px;
    padding: 2px;
}