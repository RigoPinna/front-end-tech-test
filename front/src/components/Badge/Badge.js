import React from 'react'
import PropTypes from 'prop-types'
import './Badge.css'

export const Badge = ({ number }) => {
    return (
        <span className="__badge animate__animated animate__fadeIn">
            { number }
        </span>
    )
}

Badge.propTypes = {
    number: PropTypes.number.isRequired,
}
