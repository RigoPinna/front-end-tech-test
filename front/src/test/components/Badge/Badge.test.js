import React from 'react'
import { mount } from 'enzyme'
import { Badge } from '../../../components/Badge/Badge'

describe('Pruebas en el componente <Badge />',()=> {
    test('Debe de mostrarse correctamente', () => {
        const wrapper = mount(<Badge number={1}/>)
        expect( wrapper).toMatchSnapshot();
    })
    test('Debe de mostrar el nÃºmero correctamente', () => {
        const number = 2
        const wrapper = mount(<Badge number={2}/>)
        const textNumbered = wrapper.find('span')
        expect(  +textNumbered.text() ).toBe( number )
    })
    
})