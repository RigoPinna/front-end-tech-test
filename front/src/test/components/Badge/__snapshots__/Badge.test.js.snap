// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Badge /> Debe de mostrarse correctamente 1`] = `
<span
  className="__badge animate__animated animate__fadeIn"
>
  1
</span>
`;
